datadog/resource_datadog_integration_aws.go

```diff
@@ -175,6 +183,13 @@ func buildDatadogIntegrationAwsStruct(d *schema.ResourceData) *datadogV1.AWSAcco
 		iaws.SetCspmResourceCollectionEnabled(vBool)
 	}
 
+	if d.GetRawConfig().GetAttr("extended_resource_collection_enabled").IsNull() && iaws.GetCspmResourceCollectionEnabled() {
+		iaws.SetExtendedResourceCollectionEnabled(true)
```

Comment:
Do we need this case?

Reply:
We need to check the raw config because Terraform has no way of differentiating between a unset value being interpreted as false and an explicit false value. Without this case, a user who enables `cspm` on an existing resource will get 400s because terraform thinks they're explicitly setting `extended` to false as well, when it's actually the computed value being carried forward from the previous state.